pkg/networkservice/connectioncontext/dnscontext/client.go

```diff
@@ -169,3 +165,56 @@ func (c *dnsContextClient) initialize() {
 
 	c.updateCorefileQueue.AsyncExec(c.updateCorefile)
 }
+
+func removeDuplicates(elements []*networkservice.DNSConfig) []*networkservice.DNSConfig {
+	var result []*networkservice.DNSConfig
+
+	for _, e := range elements {
+		exists := false
+		for _, v := range result {
+			if intersect(e.DnsServerIps, v.DnsServerIps) && intersect(e.SearchDomains, v.SearchDomains) {
+				exists = true
+				v.DnsServerIps = merge(v.DnsServerIps, e.DnsServerIps)
+				v.SearchDomains = merge(v.SearchDomains, e.SearchDomains)
+				break
+			}
+		}
+		if !exists {
+			result = append(result, e)
+		}
+	}
+
+	return result
+}
+
+func intersect(a, b []string) bool {
+	common := make(map[string]bool, len(a))
+	for _, v := range a {
+		common[v] = true
+	}
+
+	for _, v := range b {
+		if _, ok := common[v]; ok {
+			return true
+		}
+	}
+
+	return false
+}
+
+func merge(a, b []string) []string {
+	result := a
+	common := make(map[string]bool, len(a))
+
+	for _, v := range a {
+		common[v] = true
+	}
+
+	for _, v := range b {
+		if _, ok := common[v]; !ok {
+			result = append(result, v)
+		}
+	}
+
+	return result
+}
```

Comment:
We are going a bit over-complicated here. Moreover, am not sure about merging dns configs because it could produce a ton of new unexpected states. Thus I think we need to consider about using metadata for server dnscontext.